storage/service_account/hmac/hmac_test.go

```diff
@@ -119,21 +119,22 @@ func TestDeactivateKey(t *testing.T) {
 
 func TestGetKey(t *testing.T) {
 	tc := testutil.SystemTest(t)
-	key, err := createTestKey(tc.ProjectID)
-	defer deleteTestKey(key)
-	if err != nil {
-		t.Errorf("Error in key creation: %s", err)
-	}
-
 	testutil.Retry(t, 10, 10*time.Second, func(r *testutil.R) {
+		key, err := createTestKey(tc.ProjectID)
+		defer deleteTestKey(key)
+		if err != nil {
+			t.Fatalf("Error in key creation: %s", err)
+		}
+
+		if key == nil {
```

Comment:
Add a comment that this case should not be possible, but we added it based on earlier failures.

Reply:
This should also be `r.Errorf` followed by `return`.